plugins/jobs/src/js/components/form/helpers/MetronomeJobValidators.ts

```diff
@@ -1,64 +1,108 @@
 import { i18nMark } from "@lingui/react";
+import { pipe } from "rxjs";
+
+import {
+  isObject as isObjectUtil,
+  findNestedPropertyInObject
+} from "#SRC/js/utils/Util";
+
+import { JobOutput, FormError, UcrImageKind } from "./JobFormData";
+
+type MetronomeValidators = Record<string, (formData: JobOutput) => FormError[]>;
+
+const validation = <T>(isValid: (val: T) => boolean, message: string) => (
+  path: (i: number) => string,
+  values: T[]
+) => (errors: FormError[]) =>
+  errors.concat(
+    values.flatMap(
+      (value, index) =>
+        isValid(value) ? [] : [{ message, path: path(index).split(".") }]
+    )
+  );
+
+const isBoolean = validation<boolean | undefined>(
```

Comment:
I think all these validators are a great opportunity to leverage [`unknown`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-0.html#new-unknown-top-type). I mean if I know I put in a boolean already, why check further?

Reply:
Oh noeeees, we are still on an old TS version 😢